File: 3D-DNA_chr/step1_contig2chr_agp/change_agp.pl
#!/usr/bin/perl
my $agp=shift;
open IN,$agp or die $!;
#Superscaffold_1	1	1219797	1	W	ctg053805_np5512	1004735	2224531	-
#Superscaffold_1	1219798	1220297	2	N	hic_gap	1	500	+
#Superscaffold_1	1220298	1339497	3	W	ctg082802_np5512	1	119200	+
#Superscaffold_1	1339498	1339997	4	N	hic_gap	1	500	+
#Superscaffold_1	1339998	1566670	5	W	ctg018104_np5512	1	226673	-
my $n=0;
while(<IN>){
	chomp;
	my @f=split;
	if($f[4] =~/W/){
		my $line=join("\t",@f),"\n";
		$f[0]=~s/Scaffold_//;
		my $end=$f[7]-$f[6]+1;
		print "new_ctg.$f[0].$n\t1\t$end\t1\tW\t$f[5]\t$f[6]\t$f[7]\t$f[8]\n";
		print STDERR "new.contig.$f[0].$n\t$line\n";
		$n++;
	}
}

File: 3D-DNA_chr/step1_contig2chr_agp/change_agp.py
#!/usr/bin/env python3

import sys


n = 0
with open(sys.argv[1], 'r') as agp, open(sys.argv[2], 'w') as output, open(sys.argv[3], 'w') as error:
    for line in agp:

        if line.strip() == "":
            continue

          
   
        fields = line.strip().split('\t')
        if fields[4] == 'W':
            ctg_id = 'new_ctg.{}.{}'.format(fields[0].replace('Scaffold_', ''), n)
            start = 1
            end = int(fields[7]) - int(fields[6]) + 1
            new_line = '{}\t{}\t{}\t{}\tW\t{}\t{}\t{}\t{}\n'.format(ctg_id, start, end, 1, fields[5], fields[6], fields[7], fields[8])
            output.write(new_line)
            error.write('new.contig.{}.{}\t{}\n'.format(fields[0].replace('Scaffold_', ''), n, line.strip()) + '')

            n += 1
